### PR TITLE
mafatakd bootcamp

### PR DESCRIPTION
**Pull request by murtaza**
Murtaza's bootcamp for the UWARG computer vision team.

### CODE REVIEW

File: modules/mergesort/include/merge_sort.h [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+ *  Copyright (c) 2015, Waterloo Aerial Robotics Group (WARG)
+ *  All rights reserved.
+ *
+ *  inthis software is licensed under a modified version of the BSD 3 clause license
+ *  that should have been included with this software in a file called COPYING.txt
+ *  Otherwise it is available at:
+ *  https://raw.githubusercontent.com/UWARG/computer-vision/master/COPYING.txt
+ */
+ #include <vector>
+ // @brief uses mergesort to sort the vector that is inputted
```
Comment (benjaminwinger) [inline modules/mergesort/include/merge_sort.h]: Doxygen requires that you use doc-style comments, either `/** */` such as in the file header or `///` if you want to use c++ style comments. If you don't it won't use the comment in the generated documentation.

**Review by benjaminwinger — CHANGES_REQUESTED** [review]
There are a couple necessary changes that I've outlined, but it looks good in general.

File: modules/mergesort/src/merge_sort.cpp [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    int halfLength = (int) a.size()/2;
+    std::vector<float> leftHalf(a.begin(), a.begin() + halfLength);
+    std::vector<float> rightHalf(a.begin() + halfLength, a.end());
+
+    rightHalf = merge_sort(rightHalf);
+    leftHalf = merge_sort(leftHalf);
+    a = merge(leftHalf,rightHalf);
+    return a;
+}
+// int main(){
```
Comment (mafatakd) [inline modules/mergesort/src/merge_sort.cpp]: Alright, thanks I'll make all the necessary changes.

File: modules/mergesort/include/merge_sort.h [hunk 3]
```diff
@@ -0,0 +1,19 @@
+/**
+ * @file merge_sort.h
+ * @author WARG
+ *
+ * @section LICENSE
+ *
+ *  Copyright (c) 2015, Waterloo Aerial Robotics Group (WARG)
+ *  All rights reserved.
+ *
+ *  inthis software is licensed under a modified version of the BSD 3 clause license
```
Comment (benjaminwinger) [inline modules/mergesort/include/merge_sort.h]: Didn't see this before. Just a typo, and given this is the bootcamp you can ignore this now.

**Review by benjaminwinger — APPROVED** [review]
Looks good. Merging this and will send you an invitation to Asana (the task management system we use)

**Pull request merged**